tools/scylla-nodetool.cc

```diff
@@ -81,9 +139,27 @@ const std::vector<operation_option> operation_options{
 };
 
 const std::vector<app_template::positional_option> operation_positional_options{
+    {"keyspace", bpo::value<sstring>(), "keyspace to process", 1},
+    {"tables", bpo::value<std::vector<sstring>>(), "tables to process", -1},
 };
 
 const std::vector<operation> operations{
+    {"compact",
+            "Force a (major) compaction on one or more tables",
+R"(
+Forces a (major) compaction on one or more tables. Compaction is an optimization
+that reduces the cost of IO and CPU over time by merging rows in the background.
+
+By default, major compaction runs on all the keyspaces and tables. Major
+compactions will take all the SSTables for a column family and merge them into a
+single SSTable per shard. If a keyspace is provided, the compaction will run on
+all of the tables within that keyspace. If one or more tables are provided as
+command-line arguments, the compaction will run on all tables.
+
+Fore more information, see: https://opensource.docs.scylladb.com/stable/operating-scylla/nodetool-commands/compact.html
+)",
+            {"keyspace", "tables"},
+            compact_operation},
 };
 
```

Comment:
Missing unittest, we can take the opportunity and test how it behaves with bad command lines, valid command lines but bad parameters, and valid command lines and actually check it does what it promises. It's tedious but it's an opportunity to tighten testing around nodetool.

Reply:
How do you propose we check "it does what it promises"? I was thinking about using a mock API server, directly within the test. This would allow thoroughly checking all options. Using real clusters and checking that the desired effects are achieved will get real painful real fast.

Reply:
A mock API server is fine too (provided we have tests for the API).

As for "does what it promises", that's either "reaches the correct API endpoint with the correct parameters", or "actually does the compaction as evidenced by logs or looking at sstable files".  I don't think it would be painful (no need to terabytes of data) but a mock server is fine.

Reply:
Naturally, we don't have any tests for the REST API, beyond what indirect testing it gets via nodetool usage in dtests. So testing via mock only is not going to cut it. Maybe I will add a mock test to `test/cql-pytest`, which tests all reasonable combination of cmd line options, with the help of a mock. And add integration test to dtest, checking that the "does what it promises" happens too, with a much simplified parameter coverage.

Reply:
We have `test/rest_api/`

Reply:
Oh, I didn't know that. I will check it out to see how much it covers.

Reply:
Added unittests in v2.